integrations/ttscache-s3/src/S3TtsCache.ts

```diff
@@ -0,0 +1,153 @@
+import {
+  S3Client,
+  PutObjectCommand,
+  PutObjectCommandInput,
+  HeadObjectCommand,
+  HeadObjectCommandInput,
+  GetObjectCommand,
+  GetObjectCommandInput,
+  GetObjectCommandOutput,
+} from '@aws-sdk/client-s3';
+
+import type { Credentials } from '@aws-sdk/types';
+
+import { DeepPartial } from '@jovotech/common';
+
+import { Readable } from 'stream';
+
+import { urlJoin } from 'url-join-ts';
+
+import { TtsCachePlugin, TtsCachePluginConfig, TtsData, AudioUtilities } from '@jovotech/framework';
+
+export interface S3TtsCacheConfig extends TtsCachePluginConfig {
+  credentials: Credentials;
+  bucket: string;
+  path: string;
+  baseUrl: string;
+}
+
+export type S3TtsCacheInitConfig = DeepPartial<S3TtsCacheConfig> &
+  Pick<S3TtsCacheConfig, 'credentials' | 'bucket' | 'baseUrl'>;
+
+export class S3TtsCache extends TtsCachePlugin<S3TtsCacheConfig> {
+  readonly client: S3Client;
+
+  constructor(config: S3TtsCacheInitConfig) {
+    super(config);
+
+    this.client = new S3Client({
+      credentials: this.config.credentials,
+    });
+  }
+
+  getDefaultConfig(): S3TtsCacheConfig {
+    return {
+      credentials: {
+        accessKeyId: '',
+        secretAccessKey: '',
+      },
+      bucket: '',
+      path: '',
+      baseUrl: '',
+      returnEncodedAudio: false,
+    };
+  }
+
+  async getItem(key: string, locale: string, fileExtension: string): Promise<TtsData | undefined> {
+    let command: HeadObjectCommand | GetObjectCommand;
+    const filePath = this.getFilePath(key, locale, fileExtension);
+    const fullPath = urlJoin(this.config.baseUrl, filePath);
+
+    if (this.config.returnEncodedAudio) {
+      command = this.buildGetCommand(filePath);
+    } else {
+      command = this.buildHeadCommand(filePath);
+    }
+
+    try {
+      const response = await this.client.send(command);
+
+      const body = (response as GetObjectCommandOutput).Body;
+      if (body) {
+        const result: TtsData = {
+          key,
+          fileExtension,
+          contentType: response.ContentType,
+          url: fullPath,
+          encodedAudio: await AudioUtilities.getBase64Audio(body as Readable),
+        };
+
+        return result;
+      }
+
+      if (response.ContentType) {
+        return {
+          key,
+          fileExtension,
+          contentType: response.ContentType,
+          url: fullPath,
+        };
+      }
+    } catch (error) {
+      console.log((error as Error).message);
```

Comment:
What are possible errors? Should we pass the error to the caller?

Reply:
What are your ideas on surfacing the error. Maybe a LastError property exposed on a $ttscacheS3 object? Or an OnError callback on the Plugin class? A cache error should just behave as if no cache exists.

Reply:
I agree that a cache error shouldn't cause the app to break

Reply:
Yes, it shouldn't break anything.

The error delegation needs more design and concept work and is probably something that is not relevant for this PR.